apps/ui/src/Button.tsx

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { cva, cx, type VariantProps } from 'class-variance-authority';
+import { ButtonHTMLAttributes, forwardRef, type ReactNode } from 'react';
+
+import { LoadingIndicator } from './LoadingIndicator';
+import { Slot } from '@radix-ui/react-slot';
+
+const buttonStyles = cva(['inline-flex items-center justify-center relative'], {
+	variants: {
+		variant: {
+			primary: 'bg-sui-dark text-sui-light hover:text-white border-none',
+			secondary: 'bg-gray-90 text-gray-50 hover:text-white border-none',
+			outline:
+				'bg-white border border-steel text-steel-dark hover:text-steel-darker hover:border-steel-dark active:text-steel active:border-steel disabled:border-gray-45 disabled:text-steel-dark',
+		},
+		size: {
+			md: 'px-3 py-2 rounded-md text-bodySmall font-semibold',
+			lg: 'px-4 py-3 rounded-lg text-body font-semibold',
+		},
+	},
+	defaultVariants: {
+		variant: 'primary',
+		size: 'md',
+	},
+});
+
+export interface ButtonProps
+	extends VariantProps<typeof buttonStyles>,
+		ButtonHTMLAttributes<HTMLButtonElement> {
+	loading?: boolean;
+	before?: ReactNode;
+	after?: ReactNode;
+	asChild?: boolean;
+}
+
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+	({ variant, size, loading, children, before, after, asChild, ...props }, ref) => {
+		const Comp = asChild ? Slot : 'button';
```

Comment:
If `asChild` is true, `Comp` will always be a `div` which seems not quite right

Reply:
Not sure what you mean by this?

Reply:
Updated my comment! Ignore me haha